doc/query.md

```diff
@@ -0,0 +1,545 @@
+    <strong>{ }</strong>
+    <strong>{</strong> <em>many-expressions</em> <strong>}</strong>
+
+<em>many-expressions</em> := <em>expression</em> (&quot;,&quot; <em>expression</em>)*
+
+<em>expression</em> := 
+    <em>compound-expression</em>
+    <em>comparison-expression</em>
+
+<em>compound-expression</em> := 
+    <strong>{</strong> (&quot;$and&quot; | &quot;$nor&quot; | &quot;$or&quot;) <strong>:</strong> <strong>[</strong> <em>many-expressions</em> <strong>] }</strong>  // nor not implemented
+    
+<em>comparison-expression</em> :=
+    <strong>{</strong> <em>field</em> <strong>:</strong> <strong>{</strong> <em>operator-expression</em> <strong>} }</strong>
+
+<em>negation-expression</em> := 
+    <strong>{</strong> &quot;$not&quot; <strong>:</strong> <strong>{</strong> <em>operator-expression</em> <strong>} }</strong>
+
+<em>operator-expression</em> := 
+    <em>negation-expression</em>
+    <strong>{</strong> <em>operator</em> <strong>:</strong> <em>simple-value</em> <strong>}</strong>
+    <strong>{</strong> &quot;$regex&quot; <strong>:</strong> <em>Pattern</em> <strong>}</strong>  // not implemented
+    <strong>{</strong> &quot;$mod&quot; <strong>:</strong> <strong>[</strong> <em>divisor, remainder</em> <strong>] }</strong>  // not implemented
+    <strong>{</strong> &quot;$elemMatch&quot; <strong>: {</strong> <em>many-expressions</em> <strong>} }</strong>  // not implemented
+    <strong>{</strong> &quot;$size&quot; <strong>:</strong> <em>positive-integer</em> <strong>}</strong>  // not implemented
+    <strong>{</strong> &quot;$all&quot; <strong>:</strong> <em>array-value</em> <strong>}</strong>  // not implemented
+    <strong>{</strong> &quot;$in&quot; <strong>:</strong> <em>array-value</em> <strong>}</strong>  // not implemented
+    <strong>{</strong> &quot;$nin&quot; <strong>:</strong> <em>array-value</em> <strong>}</strong>  // not implemented
+    <strong>{</strong> &quot;$exists&quot; <strong>:</strong> <em>boolean</em> <strong>}</strong>
+    <strong>{</strong> &quot;$type&quot; <strong>:</strong> <em>type</em> <strong>}</strong>  // not implemented
+
+<em>operator</em> := &quot;$gt&quot; | &quot;$gte&quot; | &quot;$lt&quot; | &quot;$lte&quot; | &quot;$eq&quot; | &quot;$neq&quot;
+
+// Obviously List, but easier to express like this
+<em>array-value</em> := <strong>[</strong> simple-value (&quot;,&quot; simple-value)+ <strong>]</strong>
+
+// Java mappings of basic types
+
+<em>field</em> := <em>String</em>  // a field name
+
+<em>simple-value</em> := <em>String</em> | <em>Number</em>
+
+<em>positive-integer</em> := <em>Integer</em>
+
+<em>boolean</em> := <em>Boolean</em>
+
+<em>type</em> := <em>Class</em>
+</pre>
+
+
+
+## Requirements
+
+This package as well as all of its dependencies co-exist in the [Cloudant Sync - Android Datastore][2].  Installing that package by default installs this sub-package.
+
+## License
+
+Cloudant Query - Android is available under the Apache V2 license. See the LICENSE file for more info.
```

Comment:
Removed in https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3